docs/architecture/ADR-004-QGB_relayer_security.md

```diff
@@ -0,0 +1,104 @@
+
+- Cons:
+  - Unifiying the `ValsetConfirm`s and `DataCommitmentConfirm`s under the same nonce even if they represent separate concepts.
+- Pros:
+  - Simpler QGB smart contract
+
+### Add more state to the contract : Store valsets and their nonce
+
+Update the QGB contract to store the valset hashes + their nonces:
+
+- Cons:  
+		- Would make the contract more complex  
+- Pros:  
+		- Would make the relayer parallelizable (can submit data commitments and valsets in any order as long as the valset is committed)  
+		- would allow the QGB to catchup correctly even in the existence of a malicious relayer  
+
+### A request oriented design
+
+Currently, the attestations that need to be signed are defined by the state machine based on `abci.endBlock()` and a `DataCommitmentWindow`. This simplifies the state machine and doesn't require  implementing new transaction types to ask for orchestrators signatures.
+
+The request oriented design means providing users (relayers mainly) with the ability to post data commitment requests and ask orchestrators to sign them.
+
+The main issue with this approach is spamming and state bloat. In fact, allowing attestations signatures' requests would allow anyone to spam the network with unnecessary signatures and make orchestrators do unnecessary work. This gets worse if signatures are part of the state, since this latter is costly.
+
+A proposition to remediate the issues described above is to make the signatures part of the block data in specific namespaces. Then, we can charge per request and even make asking for attestations signatures a bit costly.
+
+- Pros
+	- Would allow anyone to ask for signatures over commitments, ie, the QGB can then be used by any team without changing anything.
+
+- Cons
+	- Makes slashing more complicated. In fact, to slash for liveness, providing the whole history of blocks, proving that a certain orchestrator didn't sign an attestation in the given period, will be hard to implement and the proofs will be big. Compared to the attestations being part of the state, which can be queried easilly.
+
+## Decision
+
+the **Synchronous QGB : Universal nonce approach** will be implemented as it will allow us to ship a working QGB 1.0 version faster while preserving the same security assumptions at the expense of parallelization, and custumization, as discussed under the _request oriented design_ above.
+
+## Detailed Design
+
+The implementation will mainly consist of:
+
+- Updating the state machine to output valset requests with the correct nonce
+- Add state machine tests
+- Update the orchestrator to generate the right nonce for data commitments 
+- Update the relayer implementation to be sequential
```

Comment:
Considering the the design changes that we went with is relatively small compared to some of the other options, I'm perfectly fine with not filling in all of these details now. In fact, I think this creates a fantastic opportunity for a future contributor #471 to dig through the code and figure out how it actually works.